### PR TITLE
Add `_get_default_pip_requirements` to all flavors

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

## What changes are proposed in this pull request?

Add the `_get_default_pip_requirements` fucntion to all flavors (except `xgboost`).

## How is this patch tested?

Existing checks.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/catboost.py [hunk 1]
```diff
@@ -39,17 +39,18 @@
...
     :return: The default Conda environment for MLflow Models produced by calls to
              :func:`save_model()` and :func:`log_model()`.
     """
-    import catboost as cb
-
-    return _mlflow_conda_env(
-        # CatBoost is not yet available via the default conda channels, so we install it via pip
-        additional_pip_deps=["catboost=={}".format(cb.__version__)]
-    )
+    return _mlflow_conda_env(additional_pip_deps=_get_default_pip_requirements())
```
Comment (harupy) [inline mlflow/catboost.py:53]: What this PR does is extract the code to prepare pip requirements from `get_default_conda_env` and redefine it as `_get_default_pip_requirements`. That's it.

File: mlflow/fastai.py [hunk 2]
```diff
@@ -43,6 +43,18 @@
 FLAVOR_NAME = "fastai"
 
 
+def _get_default_pip_requirements(include_cloudpickle=False):
```
Comment (harupy) [inline mlflow/fastai.py:46]:
```suggestion
def _get_default_pip_requirements(include_cloudpickle=False):
     """
     Returns a default set of pip requirements of this flavor.
     """
```

Comment (harupy) [reply mlflow/fastai.py:46]: docstring may clarify what this function returns.

**Review by dbczumar — COMMENTED** [review]
@harupy Do we need to do this if we're going to start inferring requirements?

**Comment by harupy** [comment]
@dbczumar I think we do. We can fallback to what `_get_default_pip_requirements` returns when the inference procedure fails like this:

https://github.com/mlflow/mlflow/pull/4518/files#diff-f6bd607049d4969297151a812f447bdc13453a856df017983f7ea81470d25bd0R191-R193

```python
        default_pip_requirements = infer_pip_requirements(
            path, FLAVOR_NAME, fallback=_get_default_pip_requirements()
        )
```

File: mlflow/catboost.py [hunk 3]
```diff
@@ -39,17 +39,18 @@
 _MODEL_BINARY_FILE_NAME = "model.cb"
 
 
+def _get_default_pip_requirements():
```
Comment (dbczumar) [inline mlflow/catboost.py:42]: shall we make this a public function that users can reference? (and make a similar change for other flavors)

File: mlflow/fastai.py (hunk 2 above)
Comment (dbczumar) [reply mlflow/fastai.py:46]: How about 


```
:return: A list of default pip requirements for MLflow Models produced by this flavor.
             Calls to :func:`save_model()` and :func:`log_model()` produce a pip environment
             that, at minimum, contains these requirements.
```

Comment (dbczumar) [reply mlflow/fastai.py:46]: When we implement pip dependency inference, can we also update `get_default_conda_env()` to indicate that the default conda env is a subset of the full set of dependencies generated by `save_model()` and `log_model()`? - we can do this as part of a follow-up PR.

**Review by dbczumar — APPROVED** [review]
LGTM with a couple of small comments!

File: mlflow/catboost.py (hunk 3 above)
Comment (harupy) [reply mlflow/catboost.py:42]: sure!

File: mlflow/fastai.py (hunk 2 above)
Comment (harupy) [reply mlflow/fastai.py:46]: > How about ...

Sounds good!

Comment (harupy) [reply mlflow/fastai.py:46]: got it!

**Pull request merged**